core/jobs.py

```diff
@@ -579,6 +598,22 @@ def _real_enqueue(cls, job_id, queue_name, additional_job_params):
 
 
 class MapReduceJobPipeline(base_handler.PipelineBase):
+    """
+       This class connects all the steps needed to perform a mapreduce job.
+       A mapreduce job splits the data in such a way that it is independent
+       and tasks on it can be performed in a parallel fashion. The job is 
+       processed and results are stored which are used for building the 
+       output of the whole job. 
+       The class has two methods: run() and finalized().Their chores are 
+       briefly explained below:
+
+       (a) run() method after processing the job sends the output of specific 
+           mapreduce job
+
+       (b)finalized() method has a null block. It contains pass keyword which 
+          indicated that nothings happens in this method.
+
```

Comment:
No need to write about every method of the class. Methods have docstrings themselves. This is redundant. Please, write only a brief description about the class in 2-3 lines or so. Ditto below everywhere else.